fe/fe-core/src/main/java/com/starrocks/catalog/Type.java

```diff
@@ -536,6 +538,27 @@ public String prettyPrint() {
      */
     protected abstract String prettyPrint(int lpad);
 
+    /**
+     * Used for Nest Type
+     */
+    public void setSelectedField(int pos, boolean needSetChildren) {
```

Comment:
plz add comments

Reply:
it's more high cohesion that create a inner class, and put `selectedFields`/`setSelectedField()`/`selectAll()`/`getSelectedFields()` in it.